polyfill/lib/time.mjs

```diff
@@ -292,20 +201,7 @@ export class Time {
     };
     const roundingIncrement = ES.ToTemporalRoundingIncrement(options, maximumIncrements[smallestUnit], false);
 
-    let { hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.DifferenceTime(
-      GetSlot(this, HOUR),
-      GetSlot(this, MINUTE),
-      GetSlot(this, SECOND),
-      GetSlot(this, MILLISECOND),
-      GetSlot(this, MICROSECOND),
-      GetSlot(this, NANOSECOND),
-      GetSlot(other, HOUR),
-      GetSlot(other, MINUTE),
-      GetSlot(other, SECOND),
-      GetSlot(other, MILLISECOND),
-      GetSlot(other, MICROSECOND),
-      GetSlot(other, NANOSECOND)
-    );
+    let { hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = calendar.timeUntil(this, other);
     ({ hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.RoundDuration(
```

Comment:
This is going to be a lot of work, but I believe RoundDuration will need to round according to the calendar's length of hours and minutes? It's fine to not block this PR on it.

Reply:
Well, wasn't the plan to keep Durations ISO-only? But I think I get your point more or less. Since `Time.p.add(duration)` interprets the duration in a calendar-specific manner, `Time.p.until` should produce a Duration based on the calendar. Let's make an issue to discuss this more in detail and land this? We can either generalize this as `calendar.durationRound` or just add the logic to the `until`/`since` methods on the calendar.